Core/src/Geometry/CuboidPortalShell.cpp

```diff
@@ -0,0 +1,301 @@
+        std::stringstream ss;
+        ss << static_cast<Face>(i);
+        throw std::runtime_error{"Invalid portal found in shell at " +
+                                 ss.str()};
+      }
+      m_volume->addPortal(portal);
+    }
+  }
+}
+
+bool SingleCuboidPortalShell::isValid() const {
+  return std::ranges::all_of(m_portals, [](const auto& portal) {
+    return portal == nullptr || portal->isValid();
+  });
+};
+
+std::string SingleCuboidPortalShell::label() const {
+  std::stringstream ss;
+  ss << "CuboidShell(vol=" << m_volume->volumeName() << ")";
+  return ss.str();
+}
+
+CuboidStackPortalShell::CuboidStackPortalShell(
+    const GeometryContext& gctx, std::vector<CuboidPortalShell*> shells,
+    AxisDirection direction, const Logger& logger)
+    : m_direction(direction), m_shells{std::move(shells)} {
+  std::tie(m_frontFace, m_backFace, m_sideFaces) =
+      CuboidVolumeBounds::facesFromAxisDirection(m_direction);
+
+  Vector3 dirVector{};
+  std::unordered_map<Face, AxisDirection> onSurfaceDirs;
+  switch (m_direction) {
+    case AxisDirection::AxisX:
+      dirVector = Vector3::UnitX();
+      onSurfaceDirs = {{NegativeZFace, AxisDirection::AxisX},
+                       {PositiveZFace, AxisDirection::AxisX},
+                       {NegativeYFace, AxisDirection::AxisY},
+                       {PositiveYFace, AxisDirection::AxisY}};
+      break;
+    case AxisDirection::AxisY:
+      dirVector = Vector3::UnitY();
+      onSurfaceDirs = {{NegativeZFace, AxisDirection::AxisY},
+                       {PositiveZFace, AxisDirection::AxisY},
+                       {NegativeXFace, AxisDirection::AxisX},
+                       {PositiveXFace, AxisDirection::AxisX}};
+      break;
+    case AxisDirection::AxisZ:
+      dirVector = Vector3::UnitZ();
+      onSurfaceDirs = {{NegativeXFace, AxisDirection::AxisY},
+                       {PositiveXFace, AxisDirection::AxisY},
+                       {NegativeYFace, AxisDirection::AxisX},
+                       {PositiveYFace, AxisDirection::AxisX}};
+      break;
+    default:
+      throw std::invalid_argument("CuboidPortalShell: Invalid axis direction");
+  }
+
+  ACTS_VERBOSE("Making cuboid stack shell in " << m_direction << " direction");
+  if (std::ranges::any_of(m_shells,
+                          [](const auto* shell) { return shell == nullptr; })) {
+    ACTS_ERROR("Invalid shell pointer");
+    throw std::invalid_argument("Invalid shell pointer");
+  }
+
+  ACTS_VERBOSE(" ~> " << label());
+
+  if (!std::ranges::all_of(
+          m_shells, [](const auto* shell) { return shell->isValid(); })) {
+    ACTS_ERROR("Invalid shell");
+    throw std::invalid_argument("Invalid shell");
+  }
+
+  std::sort(
+      m_shells.begin(), m_shells.end(),
+      [&dirVector](const auto& shellA, const auto& shellB) {
+        return (shellA->transform().translation().matrix().dot(dirVector) <
+                shellB->transform().translation().matrix().dot(dirVector));
+      });
```

Comment:
This is now the only place where `dirVector` is used. This gives an ordering of `m_shells` where the fron and back face are first (although their relative ordering is not defined) and then the side faces.

Could we make this explicit, where we just reassign this vector from the front/back/side faces explicitly?